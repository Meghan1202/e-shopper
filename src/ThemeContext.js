import React from 'react';

export const Theme = {
  light: {
    backgroundColor: 'darkgray',
    color: 'black',
  },
  dark: {
    backgroundColor: 'black',
    color: 'white',
  },
};

const ThemeContext = React.createContext(Theme.light);

export default ThemeContext;
