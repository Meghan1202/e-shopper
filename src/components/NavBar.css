.topnav {
  background-color: #333;
  overflow: hidden;
  height: 75px;
}

.topnav a.app__logo{
     float: left;
  color: #f2f2f2;
  text-align: center;
  padding: 14px;
  text-decoration: none;
  font-size: 17px;
}
.topnav a {
  float: right;
  color: #f2f2f2;
  text-align: center;
  padding: 14px;
  text-decoration: none;
  font-size: 17px;
}
.topnav a:hover {
  background-color: #ddd;
  color: black;
}
.topnav a.active {
  background-color: #4CAF50;
  color: white;
}

#basket__image{
    width: 15px;
    height: 15px;
}