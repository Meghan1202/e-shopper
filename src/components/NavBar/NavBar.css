.topnav {
  display: flex;
  background-color: #333;
  overflow: hidden;
  height: 75px;
  justify-content: space-between;
  align-items: center;
}

.topnav a.app__logo{
  color: #f2f2f2;
  text-align: center;
  padding: 14px;
  text-decoration: none;
  font-size: 17px;
}
.topnav a {
  color: #f2f2f2;
  text-align: center;
  padding: 14px;
  text-decoration: none;
  font-size: 17px;
}
.topnav a:hover {
  background-color: #ddd;
  color: black;
}
.topnav a.active {
  color: white;
}

#basket__image{
    width: 15px;
    height: 15px;
}
