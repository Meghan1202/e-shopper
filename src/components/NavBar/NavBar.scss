.topnav__light {
  display: flex;
  background-color: #333;
  overflow: hidden;
  height: 75px;
  justify-content: space-between;
  align-items: center;
  color: white;
}

.topnav__dark > a {
  text-decoration: none;
  color: white;
}

.topnav__light > a {
  text-decoration: none;
  color: black;
}

.topnav__dark {
  display: flex;
  background-color: black;
  overflow: hidden;
  height: 75px;
  justify-content: space-between;
  align-items: center;
  color: white;
}

.topnav a.app__logo {
  color: #f2f2f2;
  text-align: center;
  padding: 14px;
  text-decoration: none;
  font-size: 17px;
}

a {
  color: white;
  text-align: center;
  padding: 14px;
  text-decoration: none;
  font-size: 17px;
}

a:hover {
  background-color: #ddd;
  color: black;
}

a.active {
  color: white;
}

#basket__image {
  width: 15px;
  height: 15px;
}
