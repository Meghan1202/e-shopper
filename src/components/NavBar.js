import React from 'react';
import './NavBar.css';
import PropTypes from 'prop-types';

const NavBar = ({ items }) => (
  <div className="topnav">
    <a href="#home" className="app__logo">E-Shopper</a>
    <div>
      <a href="#allOrders">All Orders</a>
      <a className="active" href="#myBasket">
        <img id="basket__image" src="https://img.freepik.com/free-vector/empty-wicker-basket-flowers-large-birds-nest-eggs_135176-431.jpg?size=626&ext=jpg" alt="Basket" />
        My Basket
        <span>
          { items}
        </span>
      </a>
    </div>
  </div>
);

NavBar.propTypes = {
  items: PropTypes.number.isRequired,
};

export default NavBar;
