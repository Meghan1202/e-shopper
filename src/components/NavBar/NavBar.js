import React, { useContext } from 'react';
import './NavBar.scss';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import ThemeContext from '../../ThemeContext';

const NavBar = ({ items }) => {
  const theme = useContext(ThemeContext);
  return (
    <div data-testid="nav-bar" className={theme.backgroundColor === 'black' ? 'topnav__dark' : 'topnav__light'}>
      <Link className="app__logo" to="/">E-Shopper</Link>
      <br />
      <div>
        <Link to="/allOrder">All Orders</Link>
        {items > 0
          ? (
            <Link className="active" to="/cart">
              <img id="basket__image" src="https://img.freepik.com/free-vector/empty-wicker-basket-flowers-large-birds-nest-eggs_135176-431.jpg?size=626&ext=jpg" alt="Basket" />
              My Basket
              <span>
                {items}
              </span>
            </Link>
          ) : (
            <Link className="active" to="/cart">
              <img id="basket__image" src="https://img.freepik.com/free-vector/empty-wicker-basket-flowers-large-birds-nest-eggs_135176-431.jpg?size=626&ext=jpg" alt="Basket" />
              My Basket
              <span>
                {items}
              </span>
            </Link>
          )}
      </div>
    </div>
  );
};

NavBar.propTypes = {
  items: PropTypes.number.isRequired,
};

export default NavBar;
