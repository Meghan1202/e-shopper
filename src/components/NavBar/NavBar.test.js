import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import ThemeContext, { Theme } from '../../ThemeContext';
import NavBar from './NavBar';

describe(NavBar.name, () => {
  test('should match snapshot for theme light', () => {
    const { container } = render(
      <ThemeContext.Provider value={Theme.light}>
        <BrowserRouter><NavBar items={0} /></BrowserRouter>
      </ThemeContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  test('should match snapshot for theme dark', () => {
    const { container } = render(
      <ThemeContext.Provider value={Theme.dark}>
        <BrowserRouter><NavBar items={0} /></BrowserRouter>
      </ThemeContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  test('should display My basket', () => {
    render(
      <BrowserRouter><NavBar items={0} /></BrowserRouter>,
    );
    screen.getByText('My Basket');
  });
  test('should go to new route when E-shopper link is clicked', () => {
    render(<BrowserRouter><NavBar items={0} /></BrowserRouter>);
    const eshopperLink = screen.getByText('E-Shopper');
    fireEvent.click(eshopperLink);
    expect(document.location.href).toBe('http://localhost/');
  });
  test('should go to new route when My Basket link is clicked', () => {
    render(<BrowserRouter><NavBar items={0} /></BrowserRouter>);
    const myBasketLink = screen.getByText('My Basket');
    fireEvent.click(myBasketLink);
    expect(document.location.href).toBe('http://localhost/cart');
  });
  test('should go to new route when All Orders link is clicked', () => {
    render(<BrowserRouter><NavBar items={0} /></BrowserRouter>);
    const allOrderLink = screen.getByText('All Orders');
    fireEvent.click(allOrderLink);
    expect(document.location.href).toBe('http://localhost/allOrder');
  });
});
