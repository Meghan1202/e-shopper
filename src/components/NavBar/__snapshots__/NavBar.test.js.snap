// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match snapshot for theme dark 1`] = `
<div>
  <div
    class="topnav__dark"
    data-testid="nav-bar"
  >
    <a
      class="app__logo"
      href="/"
    >
      E-Shopper
    </a>
    <br />
    <div>
      <a
        href="/allOrder"
      >
        All Orders
      </a>
      <a
        class="active"
        href="/cart"
      >
        <img
          alt="Basket"
          id="basket__image"
          src="https://img.freepik.com/free-vector/empty-wicker-basket-flowers-large-birds-nest-eggs_135176-431.jpg?size=626&ext=jpg"
        />
        My Basket
        <span>
          0
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`NavBar should match snapshot for theme light 1`] = `
<div>
  <div
    class="topnav__light"
    data-testid="nav-bar"
  >
    <a
      class="app__logo"
      href="/"
    >
      E-Shopper
    </a>
    <br />
    <div>
      <a
        href="/allOrder"
      >
        All Orders
      </a>
      <a
        class="active"
        href="/cart"
      >
        <img
          alt="Basket"
          id="basket__image"
          src="https://img.freepik.com/free-vector/empty-wicker-basket-flowers-large-birds-nest-eggs_135176-431.jpg?size=626&ext=jpg"
        />
        My Basket
        <span>
          0
        </span>
      </a>
    </div>
  </div>
</div>
`;
